app/controllers/users_controller.rb

```diff
@@ -11,19 +11,22 @@ def new
   def create
     @user = User.new user_params
     if @user.save
-      log_in @user
-      flash[:success] = t ".welcome_messages", user_name: @user.name
-      redirect_to @user
+      @user.send_activation_email
+      flash[:info] = t(".check_your_email")
+      redirect_to root_url
     else
       render :new
     end
   end
 
   def index
-    @users = User.paginate page: params[:page], per_page: Settings.per_page
+    @users = User.where(activated: true).paginate(page: params[:page],
```

Comment:
Chú tìm hiểu về `scope in rails`; sự khác biệt giữa scope và class method

Sau đó, viết scope cho code `.where(activated: true).`

Reply:
scope khác biệt với class method ở chỗ nó có thể nối liên tiếp đc nhiều scope với nhau...ví dụ để find users với 2 scope là : activated và admin...đơn giản chỉ là User.activated.admin . kết quả trả về sẽ là những record đã active và là admin.  (tính chainable)
thứ nữa đó là scope sẽ trả về giá trị không nil...scope sẽ tự động kiểm tra kết quả trả về ...nếu kết quả trả về nil...scope sẽ thực hiện câu query : 
SELECT users.* FROM users

Reply:
class method ko thể gọi nối nhau dc ah?

thêm về sự khác nhau của kiểu dữ liệu trả về nữa nhé

Reply:
scope bản chất nó cũng là 1 class method. đơn cử thì scope chỉ thường dùng để truy vấn và lấy các đối tượng object....và trừ nil và false ra. Nếu nó trả về nil hoặc false, thay vào đó, tất cả scope sẽ được trả về. 
Em tìm mãi mà k thấy còn gì khác nữa không? Mong chị chỉ giáo cho em! :D

Reply:
method thì nó có thể trả về object, string, int ....

nhưng scope chỉ trả về object nên `class method ko thể gọi nối nhau dc ah?` -> có thể đúng hoặc sai, nếu int thì sai, trả về object thì dc :D